ntbs-service/DataAccess/DataQualityRepository.cs

```diff
@@ -102,6 +104,39 @@ public async Task<IList<Notification>> GetNotificationsEligibleForDotVotAlerts()
                 .ToListAsync();
         }
 
+        public async Task<IList<NotificationAndDuplicateIds>> GetNotificationIdsEligibleForPotentialDuplicateAlerts()
+        {
+            return await GetBaseNotificationQueryableForAlerts()
+                .SelectMany(a => _context.Notification, 
+                    (notification, duplicate) => new {notification, duplicate})
+                .Where(t =>
+                    (t.notification.PatientDetails.GivenName == t.duplicate.PatientDetails.FamilyName &&
+                     t.notification.PatientDetails.FamilyName == t.duplicate.PatientDetails.GivenName &&
+                     t.notification.PatientDetails.NhsNumber == t.duplicate.PatientDetails.NhsNumber &&
+                     t.notification.PatientDetails.NhsNumber != null &&
+                     t.notification.PatientDetails.GivenName != null &&
+                     t.notification.PatientDetails.GivenName != null) 
+                    ||
+                    (t.notification.PatientDetails.NhsNumber == t.duplicate.PatientDetails.NhsNumber &&
+                     t.notification.PatientDetails.Dob == t.duplicate.PatientDetails.Dob && 
+                     t.notification.PatientDetails.NhsNumber != null &&
+                     t.notification.PatientDetails.Dob != null)
```

Comment:
Hmmm, not sure about the logic here...
Firstly, I think it'd be useful if we put the words of the conditions as a comment next to each logical block, to make the intention explicit.
Secondly, we don't seem to cater for the case where names match in a non-reversed way!
Lastly, I don't think this was an explicit request on the ticket, but I don't think it makes sense to consider draft notifications? And if we don't, then we can get rid of the `null` checks, as these are all required fields!